src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -396,6 +427,14 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
 
     case PMake(typ, _) => typeSymbType(typ)
 
+    case PPredConstructor(base, args) =>
+      base match {
+        case PFPredBase(id) =>
+          val idT = idType(id)
+          PredT(idT.asInstanceOf[FunctionT].args diff args.filter(_.isDefined).map(x => exprType(x.get)))
```

Comment:
Is `diff` the right thing to do here? As diff takes the multiset difference between the function arguments and the applied arguments I think there might be problems if the same type occurs multiple times on the left or right

Reply:
Yeah, you should probably zip and then collect.